Source/Core/DolphinWX/Debugger/AssemblerEntryDialog.h

```diff
@@ -0,0 +1,32 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <wx/textdlg.h>
+
+#include "Common/CommonTypes.h"
+
+class wxStaticText;
+
+class AssemblerEntryDialog : public wxTextEntryDialog
+{
+public:
+  AssemblerEntryDialog(u32 address, wxWindow* parent, const wxString& message,
+                       const wxString& caption = wxGetTextFromUserPromptStr,
+                       const wxString& value = wxEmptyString, long style = wxTextEntryDialogStyle,
+                       const wxPoint& pos = wxDefaultPosition);
+
+  bool Create(wxWindow* parent, const wxString& message,
+              const wxString& caption = wxGetTextFromUserPromptStr,
+              const wxString& value = wxEmptyString, long style = wxTextEntryDialogStyle,
+              const wxPoint& pos = wxDefaultPosition);
+
+protected:
```

Comment:
I used protected instead of private in case it gets inherited to support another assembler later. I thought about DSP but that's the only thing that came to my mind.